### PR TITLE
fix(Cloud Databases):Incorporate new client-error-toolchain in data sources

### PR DESCRIPTION
**Pull request by lornakelly**
Currently we are using an outdated way of deal with errors

This PR will:
- Update our data sources to use the new client-error-toolchain
- Fix for a hardcoded location in a test

### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for pull request followers and do not help prioritize the request

Relates OR Closes #0000

Output from acceptance testing:

```
$ make testacc TESTARGS='-run=TestAccIBMDatabaseBackupDataSourceBasic'
--- PASS: TestAccIBMDatabaseBackupDataSourceBasic (18.64s)

$ make testacc TESTARGS='-run=TestAccIBMDatabaseBackupsDataSourceBasic'
--- PASS: TestAccIBMDatabaseBackupsDataSourceBasic (14.93s)

$ make testacc TESTARGS='-run=TestAccIBMDatabasePointInTimeRecoveryDataSourceBasic'
--- PASS: TestAccIBMDatabasePointInTimeRecoveryDataSourceBasic (638.15s)

$ make testacc TESTARGS='-run=TestAccIBMDatabaseTasksDataSourceBasic'
--- PASS: TestAccIBMDatabasePointInTimeRecoveryDataSourceBasic (638.15s)

$ make testacc TESTARGS='-run=TestAccIBMDatabaseDataSource_basic'
--- PASS: TestAccIBMDatabaseDataSource_basic (681.04s)

$ make testacc TESTARGS='-run=TestAccIBMDatabaseConnectionDataSourceBasic'
--- PASS: TestAccIBMDatabaseConnectionDataSourceBasic (696.35s)

...
```

### CODE REVIEW

File: ibm/service/database/data_source_ibm_database_backups.go [hunk 1]
```diff
@@ -146,19 +151,21 @@ func DataSourceIBMDatabaseBackupsRead(context context.Context, d *schema.Resourc
...
-		return diag.FromErr(fmt.Errorf("Error setting backups %s", err))
+		tfErr := flex.TerraformErrorf(err, fmt.Sprintf("Error setting backups: %s", err), "(Data) ibm_database_backups", "read")
+		return tfErr.GetDiag()
 	}
 
 	return nil
 }
 
-// DataSourceIBMDatabaseBackupsID returns a reasonable ID for the list.
+// dataSourceIBMDatabaseBackupsID returns a reasonable ID for the list.
```
Comment (omaraibrahim) [inline ibm/service/database/data_source_ibm_database_backups.go:168]: We want to capitalize this

Comment (lornakelly) [reply ibm/service/database/data_source_ibm_database_backups.go]: Updated

**Review by omaraibrahim — APPROVED** [review]
LGTM

File: go.mod [hunk 2]
```diff
@@ -8,7 +8,7 @@ require (
 	github.com/IBM/apigateway-go-sdk v0.0.0-20210714141226-a5d5d49caaca
 	github.com/IBM/appconfiguration-go-admin-sdk v0.3.0
 	github.com/IBM/appid-management-go-sdk v0.0.0-20210908164609-dd0e0eaf732f
-	github.com/IBM/cloud-databases-go-sdk v0.6.0
+	github.com/IBM/cloud-databases-go-sdk v0.7.0
```
Comment (obai-1) [inline go.mod:11]: Has this sdk version released?

Comment (lornakelly) [reply go.mod:11]: Yep https://github.com/IBM/cloud-databases-go-sdk/releases/tag/v0.7.0

**Pull request merged**